src/transformers/trainer.py

```diff
@@ -2516,7 +2517,6 @@ def _inner_training_loop(
                 self.control.should_training_stop = True
 
             self.control = self.callback_handler.on_epoch_end(args, self.state, self.control)
-            self._maybe_log_save_evaluate(tr_loss, grad_norm, model, trial, epoch, ignore_keys_for_eval)
```

Comment:
I think that you are deleting the wrong one. There is another `_maybe_log_save_evaluate` above. Could you add a test to check if the lr logged is the correct one ?